### PR TITLE
Make raw partition read work

### PR DESCRIPTION
**Pull request by zhexuany**
This is the third PR to support partition read.

Changes in this PR:
* creating ranges by partition id inserted of table id when partition table is enabled. 
 ** for index part, we only have the local index. The global index does not exist since the only pid was encoded in key. 

This PR only supports full table scan, partition pruning will be fixed in the following PRs.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala [hunk 1]
```diff
@@ -34,8 +34,14 @@ class IssueTestSuite extends BaseTiSparkSuite {
...
       """.stripMargin)
+    tidbStmt.execute(("insert into partition_t values (1, `dede1`, 1989-01-01)"))
+    tidbStmt.execute(("insert into partition_t values (2, `dede2`, 1994-01-01)"))
+    tidbStmt.execute(("insert into partition_t values (3, `dede3`, 1996-01-01)"))
+    tidbStmt.execute(("insert into partition_t values (4, `dede4`, 1997-01-01)"))
+    tidbStmt.execute(("insert into partition_t values (5, `dede5`, 2001-01-01)"))
+    tidbStmt.execute(("insert into partition_t values (6, `dede6`, 2006-01-01)"))
     refreshConnections()
-    assert(spark.sql("select * from partition_t").count() == 0)
+    assert(spark.sql("select * from partition_t").count() == 6)
```
Comment (birdstorm) [inline core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala]: should add more test cases concerning filters, groupBy, and joins.

**Comment by birdstorm** [comment]
Would you please add description stating that after this PR, requests concerning partition tables will be equivalent to full table scan plus filters.

File: tikv-client/src/main/java/com/pingcap/tikv/predicates/ScanAnalyzer.java [hunk 2]
```diff
@@ -193,60 +195,122 @@ public ScanPlan buildScan(
         keyRanges, result.getResidualPredicates(), index, cost, isDoubleRead, estimatedRowCount);
   }
 
+  Pair<Key, Key> buildTableScanKeyRangeHelper(long id, IndexRange ir) {
```
Comment (ilovesoup) [inline tikv-client/src/main/java/com/pingcap/tikv/predicates/ScanAnalyzer.java]: Why package level accessible? Give it a better name also.

**Pull request merged**